### PR TITLE
docs: add docs for app data file endpoints

### PR DESCRIPTION
**Pull request by pyphilia**
ref #1528

### CODE REVIEW

File: src/services/item/plugins/app/appData/plugins/file/schema.ts [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
 
-const download = {
+export const download = {
+  operationId: 'downloadAppDataFile',
+  tags: ['app', 'app-data', 'file'],
+  summary: 'Download app data file',
+  description: 'Download app data file.',
+
   params: customType.StrictObject({
     id: customType.UUID(),
```
Comment (spaenleh) [inline src/services/item/plugins/app/appData/plugins/file/schema.ts:30]: Same comment here, can we specify which `id` this is ?

File: src/services/item/plugins/app/appData/plugins/file/schema.ts [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
 
-const upload = {
-  querystring: {
-    type: 'object',
-    properties: {
-      id: customType.UUID(),
-    },
-    additionalProperties: false,
+  querystring: customType.StrictObject({
+    id: customType.UUID(),
```
Comment (spaenleh) [inline src/services/item/plugins/app/appData/plugins/file/schema.ts:15]: Also here.

File: src/services/item/plugins/app/appData/plugins/file/index.ts [hunk 3]
```diff
@@ -77,7 +77,7 @@ const basePlugin: FastifyPluginAsync<GraaspPluginFileOptions> = async (fastify,
   };
   appDataService.hooks.setPreHook('patch', patchPreHook);
 
-  fastify.route<{ Querystring: IdParam; Body: unknown }>({
+  fastify.route({
     method: HttpMethod.Post,
```
Comment (spaenleh) [inline src/services/item/plugins/app/appData/plugins/file/index.ts:81]: Same question here about the use of `fastify.route` instead of the shorthand.

**Review by spaenleh — CHANGES_REQUESTED** [review]
Nice ! I think we should change some things before merging. 📜

File: src/services/item/plugins/app/appSetting/plugins/file/index.ts [hunk 4]
```diff
@@ -88,7 +88,7 @@ const basePlugin: FastifyPluginAsync<GraaspPluginFileOptions> = async (fastify,
   };
   appSettingService.hooks.setPreHook('patch', patchPreHook);
 
-  fastify.route<{ Body: unknown }>({
+  fastify.route({
     method: HttpMethod.Post,
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply src/services/item/plugins/app/appSetting/plugins/file/index.ts:92]: Not sure, but it was not the goal of the PR so I didn't change it.

**Review by spaenleh — APPROVED** [review]

File: src/services/item/plugins/app/appData/plugins/file/schema.ts [hunk 5]
```diff
@@ -1,29 +1,41 @@
...
-    },
-    additionalProperties: false,
+  querystring: customType.StrictObject({
+    replyUrl: Type.Boolean({
+      default: false,
+    }),
+  }),
+
+  response: {
+    [StatusCodes.OK]: Type.String({ format: 'uri' }),
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply src/services/item/plugins/app/appData/plugins/file/schema.ts:38]: I think it does, it is for regular files or thumbnails that it doesn't.

https://github.com/graasp/graasp/blob/908a278f1116f8b34bf34ba1a9463f5ffccc3dcf/src/services/item/plugins/app/appData/plugins/file/service.ts#L96

**Pull request merged**